src/index.ts

```diff
@@ -26,33 +27,61 @@ export default class CAFReceiver {
 
   // Setup DRM if present in `media.customData`
   private readonly onLoad = (loadRequestData: LoadRequestData): LoadRequestData => {
-    if (loadRequestData.media.customData && loadRequestData.media.customData.drm) {
-      return this.setDRM(loadRequestData);
+    const customData = loadRequestData.media.customData as CAFMediaInfoCustomData;
+
+    if (customData) {
+      if (customData.options) {
+        this.setWithCredentials(customData.options);
+      }
+
+      if (customData.drm) {
+        this.setDRM(customData.drm);
+      }
     }
 
     return loadRequestData;
-  }
+  };
 
-  private setDRM(loadRequestData: LoadRequestData): LoadRequestData {
-    const protectionSystem = loadRequestData.media.customData.drm.protectionSystem;
-    const licenseUrl = loadRequestData.media.customData.drm.licenseUrl;
+  private setDRM({ protectionSystem, licenseUrl, headers, withCredentials }: CAFDrmConfig): void {
     this.context.getPlayerManager().setMediaPlaybackInfoHandler((_loadRequest, playbackConfig) => {
       playbackConfig.licenseUrl = licenseUrl;
-      playbackConfig.protectionSystem =  protectionSystem;
+      playbackConfig.protectionSystem = protectionSystem as ContentProtection;
 
-      if (typeof loadRequestData.media.customData.drm.headers === 'object') {
-        playbackConfig.licenseRequestHandler = requestInfo => {
-          requestInfo.headers = loadRequestData.media.customData.drm.headers;
+      if (typeof headers === 'object') {
+        playbackConfig.licenseRequestHandler = (requestInfo) => {
+          requestInfo.headers = headers;
         };
       }
 
+      if (withCredentials) {
+        playbackConfig.licenseRequestHandler = setWithCredentialsFlag;
+      }
+
       return playbackConfig;
     });
+  }
 
-    return loadRequestData;
+  private setWithCredentials(options: CAFSourceOptions): void {
+    const playerManager = this.context.getPlayerManager();
+    const playbackConfig = Object.assign(new cast.framework.PlaybackConfig(), playerManager.getPlaybackConfig());
+
+    if (options.withCredentials) {
+      playbackConfig.manifestRequestHandler = setWithCredentialsFlag;
+    }
+
+    if (options.manifestWithCredentials) {
+      playbackConfig.segmentRequestHandler = setWithCredentialsFlag;
+      playbackConfig.captionsRequestHandler = setWithCredentialsFlag;
+    }
```

Comment:
shouldn't this set the manifestRequesthandler? And the content of this `if` goes into `options.withCredentials`?

Reply:
Yeah, it looks to me as well that the content of this if should be in the `withCredentials` along with `manifestRequestHandler` and this if just have `manifestRequestHandler`. Or we are missing something? 🤔

Reply:
Good catch! I just did some tests on Chromecast and noticed that too.